### PR TITLE
Scheduling tests

### PR DESCRIPTION
**Pull request by kvyh**
I have written tests for various parts of the scheduling as well as fixed a couple of errors that showed up as I was testing.

### CODE REVIEW

**Comment by bmorris3** [comment]
Hey @eteq, are you familiar with this kind of failure, which seems to stem from unit comparison, and is unique to [the numpy 1.7 build](https://travis-ci.org/astropy/astroplan/jobs/149552934)?

File: astroplan/tests/test_scheduling.py [hunk 1]
```diff
@@ -21,70 +22,184 @@
...
+    constraints = [AtNightConstraint(), MoonIlluminationConstraint(max=0)]
+    scheduler1 = SequentialScheduler(start_time, end_time, constraints, apo,
+                                     transitioner)
+    schedule1 = scheduler1(block)
+    assert len(schedule1.observing_blocks) == 0
+    scheduler2 = PriorityScheduler(start_time, end_time, constraints, apo,
+                                   transitioner)
+    schedule2 = scheduler2(block)
+    assert len(schedule2.observing_blocks) == 0
+'''
```
Comment (bmorris3) [inline astroplan/tests/test_scheduling.py]: Is this waiting on another PR?

Comment (kvyh) [inline astroplan/tests/test_scheduling.py]: Yes, #213, without the speedup of the `MoonIlluminationConstraint` it takes really long.

**Comment by kvyh** [comment]
This is completely replaced by #218 and #223

**Pull request closed without merging**